modules/covid_vaccine/app/services/covid_vaccine/v0/facility_suggestion_service.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'lighthouse/facilities/client'
+require 'sentry_logging'
+require 'common/exceptions/bad_gateway'
+require 'common/exceptions/unprocessable_entity'
+
+module CovidVaccine
+  module V0
+    class FacilitySuggestionService
+      include SentryLogging
+
+      # List of VA "consolidated" facilities, VAMCs that had their
+      # VistA instances merged and therefore are VAMCs that may have a
+      # longer than 3 digit station ID
+      CONSOLIDATED_FACILITIES = %w[528 528A4 528A5 528A6 528A7 528A8
+                                   549 589 589A4 589A5 589A6 589A7
+                                   620 620A4 626 626A4 636 636A6
+                                   636A8 657 657A4 657A5].freeze
+
+      # return a list of nearby facilities based on provided zipcode
+      def facilities_for(zipcode, count = 3)
+        zcta_row = ZCTA[zipcode&.[](0...5)]
+        raise Common::Exceptions::UnprocessableEntity.new(detail: 'Invalid ZIP Code') if zcta_row.blank?
```

Comment:
@stephenBarrs @cohnjesse Will need a FE error state to handle unknown zip codes. The user should be told something along the lines of, "We were unable to recognize your zip code to offer you a facility preference. You will be registered with the medical center nearest to your given address and will be contacted by that facility for scheduling."

On the backend, these will likely go into some kind of dead letter queue for enrollment folks to handle. But TBD on that :-)

Reply:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22001